src/Voron/Data/BTrees/TreeCursor.cs

```diff
@@ -48,56 +63,56 @@ public void Update(FastStack<TreePage> stack, TreePage newVal)
 
             if (oldPageNumber == newPageNumber)
             {
-                _pagesByNum[oldPageNumber] = newVal;
+                _state.PageByNum[oldPageNumber] = newVal;
                 return;
             }
 
             _anyOverrides = true;
-            _pagesByNum[oldPageNumber] = newVal;
-            _pagesByNum.Add(newPageNumber, newVal);
+            _state.PageByNum[oldPageNumber] = newVal;
+            _state.PageByNum.Add(newPageNumber, newVal);
         }
 
         public TreePage ParentPage
         {
             get
             {
                 TreePage result;
-                if (Pages.TryPeek(2, out result))
+                if (_state.Pages.TryPeek(2, out result))
                     return result;
 
                 throw new InvalidOperationException("No parent page in cursor");
             }
         }
 
-        public TreePage CurrentPage => Pages.Peek();
+        public TreePage CurrentPage => _state.Pages.Peek();
 
-        public int PageCount => Pages.Count;
+        public int PageCount => _state.Pages.Count;
 
         public void Push(TreePage p)
         {
-            Pages.Push(p);
-            _pagesByNum.Add(p.PageNumber, p);
+            _state.Pages.Push(p);
+            _state.PageByNum.Add(p.PageNumber, p);
         }
 
         public TreePage Pop()
         {
-            if (Pages.Count == 0)
+            if (_state.Pages.Count == 0)
                 throw new InvalidOperationException("No page to pop");
 
-            var p = Pages.Pop();
+            var p = _state.Pages.Pop();
 
-            var removedPrimary = _pagesByNum.Remove(p.PageNumber);
+            var removedPrimary = _state.PageByNum.Remove(p.PageNumber);
             var removedSecondary = false;
 
             if (_anyOverrides)
             {
                 var pagesNumbersToRemove = new HashSet<long>();
 
-                foreach (var page in _pagesByNum.Where(page => page.Value.PageNumber == p.PageNumber))
+                foreach (var page in _state.PageByNum.Where(page => page.Value.PageNumber == p.PageNumber))
                     pagesNumbersToRemove.Add(page.Key);
```

Comment:
Isn't this `_state.PageByNum[p.PageNumber]` ?

Reply:
No, that is why we may have overrides.